Std/Data/ByteArray.lean

```diff
@@ -127,3 +128,18 @@ def ofFn (f : Fin n → UInt8) : ByteArray where
 
 @[simp] theorem getElem_ofFn (f : Fin n → UInt8) (i) (h : i < (ofFn f).size):
     (ofFn f)[i] = f ⟨i, size_ofFn f ▸ h⟩ := get_ofFn ..
+
+/-- More efficient implementation of `ByteArray.ofFn` -/
+def ofFnAux (f : Fin n → UInt8) : ByteArray := go 0 (mkEmpty n) where
```

Comment:
```suggestion
private def ofFnAux (f : Fin n → UInt8) : ByteArray := go 0 (mkEmpty n) where
```

Reply:
Should I also add `@[specialize]`? I'm never sure when that makes a difference.